packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/hosted_ui_state_machine.dart

```diff
@@ -193,6 +196,16 @@ class HostedUiStateMachine
       ),
     );
 
+    // Clear anonymous credentials, if there were any, and fetch authenticated
+    // credentials.
+    if (_identityPoolConfig != null) {
+      await manager.clearCredentials(
+        CognitoIdentityPoolKeys(_identityPoolConfig!),
+      );
+
+      await manager.loadSession();
+    }
```

Comment:
Q: if check this at the start of the method do we still need to call `manager.loadSession()`?

Reply:
Yes, calling `loadSession` after the credentials cleared is important, since `loadSession` will see there are no AWS credentials when there are supposed to be and request them using the user pool tokens we save above.

Reply:
so if I understood the flow correctly calling `manager.storeCredentials` will store credentials and the load credential store. loading credential store does not `loadSession` and hence we need to call `loadSession` here. is that right?

Reply:
Yeah, exactly. `storeCredentials` and `clearCredentials` work directly with the credential store. `loadSession` (which calls the `FetchAuthStateMachine`) starts by calling `loadCredentialStore` and makes a decision on what needs to be refreshed based on what's presently saved.